lms/resources/lti_launch.py

```diff
@@ -29,11 +29,20 @@ def __init__(self, request):
     @cached_property
     def course(self):
         """Get the course this LTI launch based on the request's params."""
+        course_service = self._request.find_service(name="course")
 
-        return self._request.find_service(name="course").upsert_course(
+        if existing_course := course_service.get_by_context_id(
+            self._request.parsed_params["context_id"]
+        ):
+            # Keep existing `extra` instead of replacing it with the default
+            extra = existing_course.extra
+        else:
+            extra = self._new_course_extra()
```

Comment:
`new_course_extra` could be inlined here but reckon with this is more obvious which part can be "productized"